android-components/components/support/utils/src/main/java/mozilla/components/support/ktx/util/URLStringUtils.kt

```diff
@@ -143,4 +144,25 @@ object URLStringUtils {
     private fun maybeStripTrailingSlash(url: CharSequence): CharSequence {
         return url.trimEnd('/')
     }
+
+    /**
+     * Determines whether a string is http or https URL
+     */
+    fun isHttpOrHttps(url: String): Boolean {
+        return !TextUtils.isEmpty(url) && (url.startsWith("http:") || url.startsWith("https:"))
+    }
+
+    /**
+     * Determine whether a string is a valid search query URL.
+     */
+    fun isValidSearchQueryUrl(url: String): Boolean {
+        var trimmedUrl = url.trim { it <= ' ' }
+        if (!trimmedUrl.matches("^.+?://.+?".toRegex())) {
+            // UI hint url doesn't have http scheme, so add it if necessary
+            trimmedUrl = "http://$trimmedUrl"
+        }
+        val isNetworkUrl = URLUtil.isNetworkUrl(trimmedUrl)
```

Comment:
URLUtil.isNetworkUrl does the same thing as our own `isHttpOrHttps`. We could just use our own, right?

```suggestion
        val isNetworkUrl = isHttpOrHttps(trimmedUrl)
```